src/services/mongoose/users.ts

```diff
@@ -106,10 +121,28 @@ export const UsersRepository = (): IUsersRepository => {
     }
   }
 
+  const unsetDeviceIdForUser = async (id: UserId): Promise<User | RepositoryError> => {
```

Comment:
this is very (too) specific for a database layer, wdyt @bodymindarts or @dolcalmi ?

Reply:
Agree - this looks like a layer violation. Putting use case code into a repository implementation leads to very strong coupling.
There may be exceptions when an operation must be atomic (and we don't have transaction handling)- so  it might be acceptable. We do this in at least 1 case I can think of.
But I don't see any indication of that being the case here.

Reply:
probably https://github.com/GaloyMoney/galoy/pull/2071/files#diff-501a242d86bf5d6bdcab743568228d988d5a13898af716c7fae13064b6f16033R106 is similarly a layer boundary violation

cc @krtk6160

Reply:
fixed here https://github.com/GaloyMoney/galoy/pull/2071/commits/674327722dc4ba8834690dea5d32621d84e93359